JIT Testing: Improve assertAutodiffNode error message

lobstergrindset opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#50626 JIT Testing: Improve assertAutodiffNode error message**

Differential Revision: [D25932184](https://our.internmc.facebook.com/intern/diff/D25932184)

Improved the assertAutodiffNode error message to be cleaner. Grouped nodes with the same errors together and clarified the different error categories. 

Output of error message when nodes should be autodiffed: 
```
Failure in testing nodes' autodifferentiation. One or more nodes were expected to be autodiffed, but were not found in specified fusible/nonfusible DifferentiableGraph groups. 
Specifically:
  ['b'] were not in one of the DifferentiableGraphs when they were expected to be. Did you intend for these nodes to be autodiffed? If not, remove them from the list of nonfusible nodes.
  ['a'] were found in one of the FusionGroups when they were expected to be just in a DifferentiableGraph. If it was intended for these nodes to be in FusionGroups, reclassify these nodes as fusible nodes. If these nodes were not intended to be fused, your autodifferention logic might be wrong.
  ['d'] were not in one of the fusion groups of the DifferentiableGraphs when they were expected to be. They were also not found in an outer DifferentiableGraph. Did you intend for these nodes to be autodifferentiated? If not, you should remove these nodes from the test's fusible nodes. Otherwise your autodifferention logic might be wrong.
  ['c'] were not in one of the fusion groups of the DifferentiableGraphs when they were expected to be, instead they were found just in an outer DifferentiableGraph. Did you intend for these nodes to be fused? If not, you should move these nodes into the test's nonfusible nodes. Otherwise your autodifferention logic might be wrong.
```
Output of error message when nodes shouldn't be autodiffed: 
```
Failure in testing nodes' autodifferentiation. One or more nodes were not expected to be autodiffed but were found in a DifferentiableGraph or in a fusion group of a DifferentiableGraph. Did you intend for these nodes to be autodiffed? If so, change this test to expect autodifferention. 
Specifically:
  ['a'] were not expected to be in one of the DifferentiableGraphs, but appeared in a fusion group of a DifferentiableGraph. 
  ['c'] were not expected to be in one of the DifferentiableGraphs but were.
```

Commit:
JIT Testing: Improve assertAutodiffNode error message

[ghstack-poisoned]

Commit:
Update on "JIT Testing: Improve assertAutodiffNode error message"


Differential Revision: [D25932184](https://our.internmc.facebook.com/intern/diff/D25932184)

Improved the assertAutodiffNode error message to be cleaner. Grouped nodes with the same errors together and clarified the different error categories. 

Output of error message when nodes should be autodiffed: 
```
Failure in testing nodes' autodifferentiation. One or more nodes were expected to be autodiffed, but were not found in specified fusible/nonfusible DifferentiableGraph groups. 
Specifically:
  ['b'] were not in one of the DifferentiableGraphs when they were expected to be. Did you intend for these nodes to be autodiffed? If not, remove them from the list of nonfusible nodes.
  ['a'] were found in one of the FusionGroups when they were expected to be just in a DifferentiableGraph. If it was intended for these nodes to be in FusionGroups, reclassify these nodes as fusible nodes. If these nodes were not intended to be fused, your autodifferention logic might be wrong.
  ['d'] were not in one of the fusion groups of the DifferentiableGraphs when they were expected to be. They were also not found in an outer DifferentiableGraph. Did you intend for these nodes to be autodifferentiated? If not, you should remove these nodes from the test's fusible nodes. Otherwise your autodifferention logic might be wrong.
  ['c'] were not in one of the fusion groups of the DifferentiableGraphs when they were expected to be, instead they were found just in an outer DifferentiableGraph. Did you intend for these nodes to be fused? If not, you should move these nodes into the test's nonfusible nodes. Otherwise your autodifferention logic might be wrong.
```
Output of error message when nodes shouldn't be autodiffed: 
```
Failure in testing nodes' autodifferentiation. One or more nodes were not expected to be autodiffed but were found in a DifferentiableGraph or in a fusion group of a DifferentiableGraph. Did you intend for these nodes to be autodiffed? If so, change this test to expect autodifferention. 
Specifically:
  ['a'] were not expected to be in one of the DifferentiableGraphs, but appeared in a fusion group of a DifferentiableGraph. 
  ['c'] were not expected to be in one of the DifferentiableGraphs but were.
```

[ghstack-poisoned]

mruberry commented on torch/testing/_internal/common_jit.py:
```diff
@@ -146,37 +146,69 @@ def getExportImportCopy(self, m, also_test_file=True, map_location=None):
+                    "fusible nodes. If these nodes were not intended to be fused, your " \
+                    "autodifferention logic might be wrong."
+            if len(fusion_nodes_missing) > 0:
+                err_msg += f"\n  {fusion_nodes_missing} were not in one of the fusion " \
```
"fusion groups" -> "FusionGroups" for consistency

mruberry commented on torch/testing/_internal/common_jit.py:
```diff
@@ -146,37 +146,69 @@ def getExportImportCopy(self, m, also_test_file=True, map_location=None):
+                    "remove these nodes from the test's fusible nodes. Otherwise your " \
+                    "autodifferention logic might be wrong."
+            if len(fusion_nodes_in_diff) > 0:
+                err_msg += f"\n  {fusion_nodes_in_diff} were not in one of the fusion " \
```
"fusion groups" -> "FusionGroups" for consistency

mruberry commented on torch/testing/_internal/common_jit.py:
```diff
@@ -146,37 +146,69 @@ def getExportImportCopy(self, m, also_test_file=True, map_location=None):
-                "Did you intend for these nodes to be autodiffed? " \
-                "If so, change this test to expect autodifferention. " \
+            err_msg += "One or more nodes were not expected to be autodiffed " \
+                "but were found in a DifferentiableGraph or in a fusion group " \
```
"fusion group" -> "FusionGroup" for consistency

mruberry commented on torch/testing/_internal/common_jit.py:
```diff
@@ -146,37 +146,69 @@ def getExportImportCopy(self, m, also_test_file=True, map_location=None):
+            err_msg += "One or more nodes were not expected to be autodiffed " \
+                "but were found in a DifferentiableGraph or in a fusion group " \
+                "of a DifferentiableGraph. Did you intend for these nodes to be " \
+                "autodiffed? If so, change this test to expect autodifferention. " \
```
"autodifferention" -> "autodifferentiation"

mruberry commented on torch/testing/_internal/common_jit.py:
```diff
@@ -146,37 +146,69 @@ def getExportImportCopy(self, m, also_test_file=True, map_location=None):
+                    "instead they were found just in an outer DifferentiableGraph. " \
+                    "Did you intend for these nodes to be fused? If not, you should " \
+                    "move these nodes into the test's nonfusible nodes. Otherwise your " \
+                    "autodifferention logic might be wrong."
```
"autodifferention" -> "autodifferentiation"

mruberry commented on torch/testing/_internal/common_jit.py:
```diff
@@ -146,37 +146,69 @@ def getExportImportCopy(self, m, also_test_file=True, map_location=None):
+                    "They were also not found in an outer DifferentiableGraph. Did you " \
+                    "intend for these nodes to be autodifferentiated? If not, you should " \
+                    "remove these nodes from the test's fusible nodes. Otherwise your " \
+                    "autodifferention logic might be wrong."
```
"autodifferention" -> "autodifferentiation"

mruberry commented on torch/testing/_internal/common_jit.py:
```diff
@@ -146,37 +146,69 @@ def getExportImportCopy(self, m, also_test_file=True, map_location=None):
+                    "when they were expected to be just in a DifferentiableGraph. If it was " \
+                    "intended for these nodes to be in FusionGroups, reclassify these nodes as " \
+                    "fusible nodes. If these nodes were not intended to be fused, your " \
+                    "autodifferention logic might be wrong."
```
"autodifferention" -> "autodifferentiation"

mruberry commented on torch/testing/_internal/common_jit.py:
```diff
@@ -146,37 +146,69 @@ def getExportImportCopy(self, m, also_test_file=True, map_location=None):
-                    "DifferentiableGraph but was."
+            if len(fusion_nodes_found) > 0:
+                err_msg += f"\n  {fusion_nodes_found} were not expected to be in " \
+                    "one of the DifferentiableGraphs, but appeared in a fusion " \
```
"fusion group" -> "FusionGroup"

mruberry reviewed (approved):
Cool! I just made a few inline wording suggestions.

Commit:
Update on "JIT Testing: Improve assertAutodiffNode error message"


Differential Revision: [D25932184](https://our.internmc.facebook.com/intern/diff/D25932184)

Improved the assertAutodiffNode error message to be cleaner. Grouped nodes with the same errors together and clarified the different error categories. 

Output of error message when nodes should be autodiffed: 
```
Failure in testing nodes' autodifferentiation. One or more nodes were expected to be autodiffed, but were not found in specified fusible/nonfusible DifferentiableGraph groups. 
Specifically:
  ['b'] were not in one of the DifferentiableGraphs when they were expected to be. Did you intend for these nodes to be autodiffed? If not, remove them from the list of nonfusible nodes.
  ['a'] were found in one of the FusionGroups when they were expected to be just in a DifferentiableGraph. If it was intended for these nodes to be in FusionGroups, reclassify these nodes as fusible nodes. If these nodes were not intended to be fused, your autodifferention logic might be wrong.
  ['d'] were not in one of the fusion groups of the DifferentiableGraphs when they were expected to be. They were also not found in an outer DifferentiableGraph. Did you intend for these nodes to be autodifferentiated? If not, you should remove these nodes from the test's fusible nodes. Otherwise your autodifferention logic might be wrong.
  ['c'] were not in one of the fusion groups of the DifferentiableGraphs when they were expected to be, instead they were found just in an outer DifferentiableGraph. Did you intend for these nodes to be fused? If not, you should move these nodes into the test's nonfusible nodes. Otherwise your autodifferention logic might be wrong.
```
Output of error message when nodes shouldn't be autodiffed: 
```
Failure in testing nodes' autodifferentiation. One or more nodes were not expected to be autodiffed but were found in a DifferentiableGraph or in a fusion group of a DifferentiableGraph. Did you intend for these nodes to be autodiffed? If so, change this test to expect autodifferention. 
Specifically:
  ['a'] were not expected to be in one of the DifferentiableGraphs, but appeared in a fusion group of a DifferentiableGraph. 
  ['c'] were not expected to be in one of the DifferentiableGraphs but were.
```

[ghstack-poisoned]
